### PR TITLE
[SPARK-44567][INFRA] Add a new Daily testing GitHub Action job for Maven

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims add a new Daily testing GitHub Action job for Maven.

### Why are the changes needed?
Need Daily testing for Maven.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
Maven test job may have some test cases that have failed, let's fix them in followup.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -283,6 +284,179 @@ jobs:
         name: unit-tests-log-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/unit-tests.log"
 
+  # Maven Test: build Spark and run the tests for specified modules using maven.
+  maven-test:
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:288]: Hm. should we make a separate file? I think at least we can leverage `uses`

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:288]: Sorry, didn't understand what you meant.

Do you mean we shoud merge `maven-test` into an existing `build`?

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml]: Already move to a new file, but I don't know how to trigger a test for `build_and_test_maven.yml` in the current pr now ... :(

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml]: Sorry, `On pull request update / Notify test workflow ` always failed, I close this one and open a new one : https://github.com/apache/spark/pull/42200

Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml]: @Yikun  do you know how to check a new workflow?

File: .github/workflows/build_and_test_maven.yml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Build and test using Maven
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test_maven.yml:20]: I think you can call this workflow in `build_and_test.yml` by using `uses`

Comment (LuciferYang) [reply .github/workflows/build_and_test_maven.yml:20]: ok, let me try this

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (Yikun) [reply .github/workflows/build_and_test.yml]: IIUC, the workflow will be shown in personal actions page:
https://github.com/LuciferYang/spark/actions/workflows/build_main.yml

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml]: I accidentally deleted this branch, the new PR is https://github.com/apache/spark/pull/42203

**Pull request closed without merging**